lib_async/ocaml_async_kafka.c

```diff
@@ -75,8 +75,35 @@ CAMLprim value ocaml_kafka_async_new_producer(value caml_delivery_callback, valu
   CAMLreturn(result);
 }
 
-CAMLprim value ocaml_kafka_async_new_consumer(value caml_consumer_options) {
-  CAMLparam1(caml_consumer_options);
+static void ocaml_kafka_async_rebalance_cb(rd_kafka_t *rk, rd_kafka_resp_err_t err, rd_kafka_topic_partition_list_t *partitions, void *opaque) {
+  CAMLparam0();
+  CAMLlocal1(caml_cb);
+
+  caml_cb = ((ocaml_kafka_opaque*)opaque)->caml_callback;
+
+  // do the default thing, but also call our OCaml code
+  switch (err)
```

Comment:
this should probably handle cooperative assignors too, and call `rd_kafka_incremental_assign` (see https://docs.confluent.io/platform/current/clients/librdkafka/html/rdkafka_8h.html#a10db731dc1a295bd9884e4f8cb199311)